rocksdb_admin/tests/application_db_test.cpp

```diff
@@ -64,65 +77,169 @@ class ApplicationDBTestBase : public testing::Test {
     Options options;
     SstFileWriter sst_file_writer(EnvOptions(), options, options.comparator);
     auto s = sst_file_writer.Open(sst_filename);
-    EXPECT_TRUE(s.ok());
+    ASSERT_TRUE(s.ok()) << s.ToString();
 
     list<pair<string, string>>::iterator it;
     for (it = key_vals.begin(); it != key_vals.end(); ++it) {
       s = sst_file_writer.Put(it->first, it->second);
-      EXPECT_TRUE(s.ok());
+      ASSERT_TRUE(s.ok()) << s.ToString();
     }
 
     s = sst_file_writer.Finish();
-    EXPECT_TRUE(s.ok());
+    ASSERT_TRUE(s.ok()) << s.ToString();
   }
 
-  void recreateDBWithAllowIngestBehind() {
-    string name = db_name();
-    string path = db_path();
-    db_.reset(nullptr);
+  int numTableFilesAtLevel(int level) {
+    std::string p;
+    db_->rocksdb()->GetProperty("rocksdb.num-files-at-level" + std::to_string(level), &p);
+    return atoi(p.c_str());
+  }
 
-    Options options = buildDBOptions(false);
-    auto status = DestroyDB(path, options);
-    EXPECT_TRUE(status.ok());
+  int numCompactPending() {
+    uint64_t p;
+    db_->rocksdb()->GetIntProperty("rocksdb.compaction-pending", &p);
+    return static_cast<int>(p);
+  }
+
+  string levelStats() {
+    std::string p;
+    db_->rocksdb()->GetProperty("rocksdb.levelstats", &p);
+    return p;
+  }
+
+  void Reopen(const Options& options) {
+    close();
+    last_options_ = options;
+    DB* db;
+    if (FLAGS_log_to_stdout) {
+      cout << "Open DB with db_path: " << db_path_ << endl;
+    }
+    auto status = DB::Open(options, db_path_, &db);
+    ASSERT_TRUE(status.ok()) << status.ToString();
+    ASSERT_TRUE(db != nullptr);
+    db_ = make_unique<ApplicationDB>(
+        db_name_, shared_ptr<DB>(db), replicator::DBRole::SLAVE, nullptr);
```

Comment:
Will this always be SLAVE role?

Reply:
By default, a new DB is always created as "S" role. 
For all the tests now in this test file, it only operate at DB level, it does not have any replication logic, so, it does not require different roles like "M" or "S". 

If future tests need rely on the replication, ie. need both "M" and "S", then, we can add another helper function to change role.